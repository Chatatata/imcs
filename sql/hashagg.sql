select cs_hash_max(Close,Day % 2) from Quote_get('IBM');
select cs_hash_min(Close,Day % 2) from Quote_get('IBM');
select cs_hash_sum(Close,Day % 2) from Quote_get('IBM');
select cs_hash_avg(Close,Day % 2) from Quote_get('IBM');
select (q.p).agg_val,cs_cut((q.p).group_by,'i4i4') from (select cs_project_agg(cs_hash_sum(Close,(Day % 2)||(Volume%10))) p from Quote_get('IBM')) q;
select p.agg_val,cs_cut(p.group_by,'i4i4') from (select (cs_project_agg(cs_hash_sum(Close,(Day % 2)||(Volume%10)))).* from Quote_get('IBM')) p;
create type PairOfInt as (first integer, second integer);
select p.agg_val,cs_as(p.group_by,'PairOfInt') from (select (cs_project_agg(cs_hash_sum(Close,(Day % 2)||(Volume%10)))).* from Quote_get('IBM')) p;

select cs_hash_count(cs_floor((High-Low)*10)) from Quote_get('IBM');
select cs_hash_dup_count(cs_ceil((High-Low)*10), Day%3) from Quote_get('IBM');
