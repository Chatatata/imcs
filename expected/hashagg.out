select cs_hash_max(Close,Day % 2) from Quote_get('IBM');
             cs_hash_max             
-------------------------------------
 ("float4:{50.5,40.2}","int4:{0,1}")
(1 row)

select cs_hash_min(Close,Day % 2) from Quote_get('IBM');
             cs_hash_min             
-------------------------------------
 ("float4:{20.2,10.5}","int4:{0,1}")
(1 row)

select cs_hash_sum(Close,Day % 2) from Quote_get('IBM');
             cs_hash_sum              
--------------------------------------
 ("float8:{100.9,50.7}","int4:{0,1}")
(1 row)

select cs_hash_avg(Close,Day % 2) from Quote_get('IBM');
               cs_hash_avg               
-----------------------------------------
 ("float8:{33.6333,25.35}","int4:{0,1}")
(1 row)

select (q.p).agg_val,cs_cut((q.p).group_by,'i4i4') from (select cs_project_agg(cs_hash_sum(Close,(Day % 2)||(Volume%10))) p from Quote_get('IBM')) q;
     agg_val      | cs_cut 
------------------+--------
 50.7000007629395 | (1,0)
 100.900001525879 | (0,0)
(2 rows)

select p.agg_val,cs_cut(p.group_by,'i4i4') from (select (cs_project_agg(cs_hash_sum(Close,(Day % 2)||(Volume%10)))).* from Quote_get('IBM')) p;
     agg_val      | cs_cut 
------------------+--------
 50.7000007629395 | (1,0)
 100.900001525879 | (0,0)
(2 rows)

create type PairOfInt as (first integer, second integer);
select p.agg_val,cs_as(p.group_by,'PairOfInt') from (select (cs_project_agg(cs_hash_sum(Close,(Day % 2)||(Volume%10)))).* from Quote_get('IBM')) p;
     agg_val      | cs_as 
------------------+-------
 50.7000007629395 | (1,0)
 100.900001525879 | (0,0)
(2 rows)

select cs_hash_count(cs_floor((High-Low)*10)) from Quote_get('IBM');
         cs_hash_count          
--------------------------------
 ("int8:{4,1}","float8:{10,0}")
(1 row)

select cs_hash_dup_count(cs_ceil((High-Low)*10), Day%3) from Quote_get('IBM');
        cs_hash_dup_count        
---------------------------------
 ("int8:{1,1,2}","int4:{1,0,2}")
(1 row)

